File: 435-non-overlapping-intervals/non-overlapping-intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {

        sort(intervals.begin(),intervals.end(),[](const vector<int>& a, const vector<int>& b) {
         
         return a[1] < b[1]; 
});

int prevEnd=intervals[0][1];

int count=0;

for(int i=1;i<intervals.size();i++)
{
    int currEnd=intervals[i][1];
    int currStart=intervals[i][0];

    if(currStart<prevEnd)
    {
        count++;
    }
    else{
        prevEnd=currEnd;

    }
    

}

return count;

        
    }
};

File: 435-non-overlapping-intervals/non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:

        # basically in this question we have to remove the interval which ends late

        # so to do this we can first sort the intervals then the intervals with closer starts are clubbed together

        intervals.sort()


        prevEnd=intervals[0][1]
        res=0


        for start,end in intervals[1:]:
            if start>=prevEnd:
                prevEnd=end
            else:
                prevEnd=min(end,prevEnd)
                res+=1

        return res



        